pydatastructs/miscellaneous_data_structures/queue.py

```diff
@@ -0,0 +1,95 @@
+from pydatastructs.linear_data_structures import DynamicOneDimensionalArray
+
+__all__ = [
+    'Queue'
+]
+
+class Queue(object):
+    """Representation of queue data structure
+
+    Examples
+    ========
+
+    >>> from pydatastructs import Queue
+    >>> q = Queue()
+    >>> q.append(1)
+    >>> q.append(2)
+    >>> q.append(3)
+    >>> q.popleft()
+    >>> q.len()
+    1
+
+    References
+    ==========
+
+    .. [1] https://en.wikipedia.org/wiki/Queue_(abstract_data_type)
+    """
+
+    def __new__(cls, implementation='array', **kwargs):
+        if implementation == 'array':
+            return ArrayQueue(
+                kwargs.get('items', None),
+                kwargs.get('dtype', int))
+        raise NotImplementedError(
+                "%s hasn't been implemented yet."%(implementation))
+
+    def append(self, *args, **kwargs):
+        raise NotImplementedError(
+            "This is an abstract method.")
+
+    def popleft(self, *args, **kwargs):
+        raise NotImplementedError(
+            "This is an abstract method.")
+
+    @property
+    def is_empty(self):
+        return self.front == -1
+
+    @property
+    def is_full(self):
+        return ((self.rear + 1) % self.size == self.front)
+
+class ArrayQueue(Queue):
+
+    __slots__ = ['items', 'dtype']
+
+    def __new__(cls, items=None, dtype=int):
+        if items is None:
+            items = DynamicOneDimensionalArray(dtype, 0)
+        else:
+            items = DynamicOneDimensionalArray(dtype, items)
+        obj = object.__new__(cls)
+        obj.items, obj.dtype = \
+            items, items._dtype
+        return obj
+
+    def __init__(self):
+        self.items.size = size
+        self.items = [None for i in range(size)]
+        self.items.front = self.items.rear = -1
```

Comment:
Please remove this. Only `__new__` is used for creating and initialising objects.